### PR TITLE
Bump regl-scatter2d and regl-splom and maintain point-cluster at v3

### PR DESCRIPTION
**Pull request by archmoj**
Initial attempt: 
Addressing #897.
So that we could add `scattergl`, `splom` and `scatterpolargl` to `strict` partial bundles in v2.
cc: #5413

Final result:
Bump `regl-scatter2d` and `regl-splom` and maintain `point-cluster` at v3 to avoid potential impacts.

@plotly/plotly_js 
cc: @dy

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "@plotly/d3": "^3.5.18",
     "@plotly/d3-sankey": "0.7.2",
     "@plotly/d3-sankey-circular": "0.33.1",
-    "@plotly/point-cluster": "^3.1.9",
+    "@plotly/point-cluster": "git://github.com/plotly/point-cluster.git#47d99d990af3741ee30d74f42c83ba5d6d74b509",
```
Comment (archmoj) [inline package.json:63]: https://github.com/plotly/point-cluster/pull/3

**Comment by dy** [comment]
Hm, is `lod` not being used? Afaik it allowed significant perf boost for 1M+ points scatterplot interactions.

**Comment by dy** [comment]
I am not the owner of `@plotly/point-cluster` (btw it seems that fork was a tmp solution, I could transfer point-cluster to gl-vis), but ideally it would take rewriting binary-search-bounds to avoid _Function_ constructor.

File: package.json [hunk 2]
```diff
@@ -110,8 +110,8 @@
     "regl": "^1.6.1",
     "regl-error2d": "^2.0.11",
     "regl-line2d": "^3.1.0",
-    "regl-scatter2d": "^3.2.1",
-    "regl-splom": "^1.0.12",
+    "regl-scatter2d": "git://github.com/gl-vis/regl-scatter2d.git#14876b1f5af6654405055f7dede02164c052163d",
```
Comment (archmoj) [inline package.json:113]: https://github.com/gl-vis/regl-scatter2d/pull/30

File: package.json [hunk 3]
```diff
@@ -110,8 +110,8 @@
...
+    "regl-splom": "git://github.com/gl-vis/regl-splom.git#842cfedbd3a791b6b825895fae291f3906ee4f7d",
```
Comment (archmoj) [inline package.json:114]: https://github.com/gl-vis/regl-splom/pull/20

**Comment by archmoj** [comment]
> Hm, is `lod` not being used? Afaik it allowed significant perf boost for 1M+ points scatterplot interactions.

I don't see that getting called.
The hover appears to work well on tests as well as [this codepen](https://codepen.io/MojtabaSamimi/pen/ExgrePY?editors=0010).

**Comment by dy** [comment]
It's not hover but panning on zoomed part of the plot.

**Review by alexcjohnson — APPROVED** [review]
Looks like you're right, this isn't getting used. If we find any cases where performance is suffering that `lod` is expected to help, we can investigate fixing the function constructor in `binary-search-bounds`. 💃

**Comment by dy** [comment]
The difference can be seen here https://gl-vis.github.io/regl-scatter2d/ and https://gl-vis.github.io/regl-scatter2d/no-snapping
The first one is with `lod`, the second - without.
Dropping `lod` big blobs get slow on panning.

**Comment by dy** [comment]
The level of details is the core points-clustering mechanism made by @mikolalysenko long ago, without it clustering is useless, because all possible points get displayed.

**Comment by archmoj** [comment]
> The level of details is the core points-clustering mechanism made by @mikolalysenko long ago, without it clustering is useless, because all possible points get displayed.

To avoid potential impacts here as well as the `regl-scatter2d` users we maintained the version of `point-cluster` at v3 here as well as `regl-splom` and `regl-scatter2d` until https://github.com/mikolalysenko/binary-search-bounds/pull/8 could be addressed.

**Review by alexcjohnson — APPROVED** [review]
💃  @dy sorry for all the noise, we'll get the proper solution implemented eventually!

**Pull request merged**